### ISSUE TITLE
Add PublicLock v13 and Unlock v12 to unlock-js

### ISSUE THREAD

**Issue by clemsos** [unlock-protocol/unlock#11524]
We need to update the `unlock-js` package according to recent changes in the contract

## PublicLock

these functions were removed  :
- `function addKeyGranter(address account) external`
- ` function isKeyGranter(address account) external view returns (bool)`

and that one was added
-  `function isRenewable(uint256 tokenId, address referrer) external view returns (bool);`

## Unlock 

these were added

- `function setProtocolFee(uint _protocolFee) external;`  (admin, should NOT be added in Unlock JS)
- `function protocolFee() external view returns (uint);`
- `function getAdmin() external view returns (address);` (admin, should NOT be added in Unlock JS)
- `function postLockUpgrade() external;`(admin, should NOT be added in Unlock JS)

None were removed.

The only change will be the `protocolFee` getter. It is currently set to zero but should enter into price calculations at some point

for reference, here are the diffs between contract interfaces https://gist.github.com/clemsos/5e16d00c915ff629738c64040af70f91

**Comment by clemsos** [comment]
We are not parsing key granters on the subgraph so that upgrade shouldn't cause any issue there.

**Comment by julien51** [comment]
Can we show a changelog of all the git commits that are relevant @clemsos ?

**Comment by clemsos** [comment]
> Can we show a changelog of all the git commits that are relevant @clemsos ?

Yes, I added contract full diffs : https://gist.github.com/clemsos/5e16d00c915ff629738c64040af70f91

Here is the full smart-contracts git log

```
c75b0b1ac feat(smart-contracts): add Unlock V12 goerli deployment manifest (#11525)
ba798a215 feat(smart-contracts): task to submit template to multisig w/o setting as default (#11518)
491a5b8a7 fix(smart-contracts): `receive()` function safety upgrade (#11510)
39f308dee chore(smart-contracts): add @abdulhakim-altunkaya to the clabot (#11515)
cd796b0b5 deps(smart-contracts): bump OZ contracts to 4.8.2 (#11512)
eec9ff1f2 feat(smart-contracts): move template init logic into `addLockTemplate` (#11436)
6dd61b8aa fix(smart-contracts): remove unused imports in Unlock v12 (#11427)
2e569157b feat(smart-contracts): Lock upgrade hardhat script/task (#11408)
8178f4cd1 feat(smart-contracts): add `getAdmin` helper in Unlock (#11421)
fb7cdeb29 fix(smart-contracts): tests for upgrading a lock from v12 to v13 (#11407)
90ce6127b feature(smart-contracts): deployed on base's test net (#11372)
183c05954 feature(unlock-app): adding the ability to load the auth screen as modal (#11334)
16882e346 fix(smart-contracts): a lock manager is a key manager by default (#11356)
c6c9bb6aa feat(smart-contracts): bump Unlock version to 12 (#11351)
18e01bc9c fix(smart-contracts): transfers are never disabled for lock managers (#11355)
e8ab7481b feat(smart-contracts): add a `isRenewable` public function (#11347)
722443f34 feat(smart-contracts): remove `transfer` function (#11362)
2a69f97a9 fix(smart-contracts): bypass delegatecall OZ warning using decorator (#11311)
766f4f40a deps(smart-contracts): upgrade smart-contracts folder deps (#11312)
c15b2d12b fix(smart-contracts): add uniswap router WL (#11293)
f3cc1e69d fix(smart-contracts): contracts folder cleanup (#11268)
f19586344 feat(smart-contracts): add params to `isValidKey` (#11151)
81f3a5e86 fix(smart-contracts): add erc20 transfer helpers functions to save contract size (#11155)
4909747a2 fix(smart-contracts): parse ProxyAdmin address directly from contract storage (#11242)
3e1955cfb feat(smart-contracts): implement protocol fee (#11022)
59d60d031 feat(smart-contracts): emit `GasRefundValue` event (#11152)
1a7894321 feat(smart-contracts): emit `EventHooksUpdated` event (#11153)
d512ffc46 feat(smart-contracts): emit a `ReferrerFee` event (#11135)
43609c204 fix(smart-contracts): prevent referrer fee from being sent to address zero (#11136)
e8ab7481b feat(smart-contracts): add a `isRenewable` public function (#11347)
722443f34 feat(smart-contracts): remove `transfer` function (#11362)
2a69f97a9 fix(smart-contracts): bypass delegatecall OZ warning using decorator (#11311)
766f4f40a deps(smart-contracts): upgrade smart-contracts folder deps (#11312)
c15b2d12b fix(smart-contracts): add uniswap router WL (#11293)
f3cc1e69d fix(smart-contracts): contracts folder cleanup (#11268)
f19586344 feat(smart-contracts): add params to `isValidKey` (#11151)
81f3a5e86 fix(smart-contracts): add erc20 transfer helpers functions to save contract size (#11155)
4909747a2 fix(smart-contracts): parse ProxyAdmin address directly from contract storage (#11242)
3e1955cfb feat(smart-contracts): implement protocol fee (#11022)
59d60d031 feat(smart-contracts): emit `GasRefundValue` event (#11152)
1a7894321 feat(smart-contracts): emit `EventHooksUpdated` event (#11153)
d512ffc46 feat(smart-contracts): emit a `ReferrerFee` event (#11135)
43609c204 fix(smart-contracts): prevent referrer fee from being sent to address zero (#11136)
1bb69c1ea fix(smart-contracts): lower PublicLock contract size (again) (#11134)
c4c0d4a60 feat(smart-contracts): UnlockSwapPurchaser deployments (#11033)
5167f12aa feat(smart-contracts): prepare Unlock migration on mainnet (#10931)
8fb4273ea feat(smart-contracts): improve scripts ro run a task for all unlock networks (#11045)
dfb171104 fix(smart-contracts): bypass delegatecall check on OZ upgrades plugin (#10997)
4356bfc0a Revert "feat(smart-contracts): use solc 0.8.17 as latest version (#10974)"
7de31de2e feat(smart-contracts): use solc 0.8.17 as latest version (#10974)
c54da2a26 feat(smart-contracts): use local `IUniswapOracleV3` in Unlock (#10977)
0e8aa5fd7 feat(smart-contracts): use OZ native role functions to manager KeyGranters  (#10975)
b0f63c60c fix(smart-contracts): bump hardhat and solidity-coverage (#10973)
da03b302e feat(smart-contracts): settings for hardhat to use forked node properly (#10976)
c2a85b323 feat(smart-contracts): move swapAndCall to a standalone purchaser contract (#10954)
4b72bf170 fix(smart-contracts): `getBalance` test helper default to use ethersjs (#10967)
d6e18ab4d feat(smart-contracts): fix renovate smart contracts + version testing (again) (#10953)
584649f23 Revert "(smart-contracts): fix renovate PR #10459 + remove OZ upgrade… (#10947)
ac6f9d70f (smart-contracts): fix renovate PR #10459 + remove OZ upgrades plugin in Unlock versions tests (#10929)
87e7d9f7a feat(smart-contracts): bump `PublicLock` contract to v13 (#10930)
```

**Comment by clemsos** [comment]
Note that there also a few subgraph updates to perform to fully support v13 https://github.com/unlock-protocol/unlock/issues/11539
